server/src/main/java/com/linecorp/centraldogma/server/internal/api/RepositoryServiceV1.java

```diff
@@ -186,8 +191,41 @@ public CompletableFuture<RepositoryDto> patchRepository(@Param String repoName,
      */
     @Get("/projects/{projectName}/repos/{repoName}/revision/{revision}")
     @RequiresReadPermission
-    public Map<String, Integer> normalizeRevision(Repository repository, @Param String revision) {
+    public Map<String, Integer> normalizeRevision(ServiceRequestContext ctx,
+                                                  Repository repository, @Param String revision) {
         final Revision normalizedRevision = repository.normalizeNow(new Revision(revision));
+        final Revision head = repository.normalizeNow(Revision.HEAD);
+        increaseCounterIfOldRevisionUsed(ctx, repository.parent().name(), repository.name(),
+                                         normalizedRevision, head);
         return ImmutableMap.of("revision", normalizedRevision.major());
     }
+
+    static void increaseCounterIfOldRevisionUsed(ServiceRequestContext ctx, Repository repository,
+                                                 Revision revision) {
+        final Revision normalized = repository.normalizeNow(revision);
+        final Revision head = repository.normalizeNow(Revision.HEAD);
+        increaseCounterIfOldRevisionUsed(ctx, repository.parent().name(), repository.name(), normalized, head);
+    }
+
+    public static void increaseCounterIfOldRevisionUsed(
+            ServiceRequestContext ctx, String projectName, String repoName,
+            Revision normalized, Revision head) {
+        if (normalized.major() == 1 || head.major() - normalized.major() >= 5000) {
+            ctx.log().whenComplete().thenAccept(log -> {
```

Comment:
Question) Do we want to differentiate between cases using `1` vs. `>5000`?
Imagining that the counter goes up, I wonder if we want to know which case needs handling.